gcloud-java-bigquery/src/main/java/com/google/gcloud/bigquery/QueryResponse.java

```diff
@@ -0,0 +1,233 @@
+    private Page<List<FieldValue>> rows;
+    private Long totalBytesProcessed;
+    private Boolean jobComplete;
+    private List<BigQueryError> errors;
+    private Boolean cacheHit;
+
+    private Builder() {}
+
+    Builder etag(String etag) {
+      this.etag = etag;
+      return this;
+    }
+
+    Builder schema(Schema schema) {
+      this.schema = schema;
+      return this;
+    }
+
+    Builder job(JobId job) {
+      this.job = job;
+      return this;
+    }
+
+    Builder totalRows(Long totalRows) {
+      this.totalRows = totalRows;
+      return this;
+    }
+
+    Builder rows(Page<List<FieldValue>> rows) {
+      this.rows = rows;
+      return this;
+    }
+
+    Builder totalBytesProcessed(Long totalBytesProcessed) {
+      this.totalBytesProcessed = totalBytesProcessed;
+      return this;
+    }
+
+    Builder jobComplete(Boolean jobComplete) {
+      this.jobComplete = jobComplete;
+      return this;
+    }
+
+    Builder errors(List<BigQueryError> errors) {
+      this.errors = errors;
+      return this;
+    }
+
+    Builder cacheHit(Boolean cacheHit) {
+      this.cacheHit = cacheHit;
+      return this;
+    }
+
+    QueryResponse build() {
+      return new QueryResponse(this);
+    }
+  }
+
+  private QueryResponse(Builder builder) {
+    this.etag = builder.etag;
+    this.schema = builder.schema;
+    this.job = builder.job;
+    this.totalRows = builder.totalRows;
+    this.rows = builder.rows;
+    this.totalBytesProcessed = builder.totalBytesProcessed;
+    this.jobComplete = builder.jobComplete;
+    this.errors = builder.errors;
+    this.cacheHit = builder.cacheHit;
+  }
+
+  /**
+   * Returns the hash of the {@code QueryResponse} resource or {@code null} if not set.
+   */
+  public String etag() {
+    return etag;
+  }
+
+  /**
+   * Returns the schema of the results when the query completed successfully. Returns {@code null}
+   * otherwise.
+   */
+  public Schema schema() {
+    return schema;
+  }
+
+  /**
+   * Returns the identity of the BigQuery Job that was created to run the query. This field will be
+   * present even if the original request timed out.
+   */
+  public JobId job() {
+    return job;
+  }
+
+  /**
+   * Returns the total number of rows in the complete query result set, which can be more than the
+   * number of rows in the first page of results returned by {@link #rows()}. Returns {@code null}
+   * if the query did not complete successfully.
+   */
+  public Long totalRows() {
+    return totalRows;
+  }
+
+  /**
+   * Returns the query result as a paginated list of rows, if the query completed successfully.
+   * Returns {@code null} otherwise.
+   */
+  public Page<List<FieldValue>> rows() {
+    return rows;
+  }
+
+  /**
+   * Returns the total number of bytes processed for the query.
+   */
+  public Long totalBytesProcessed() {
+    return totalBytesProcessed;
+  }
+
+  /**
+   * Returns whether the job running the query has completed or not. If {@link #rows()} and
```

Comment:
I know that this is a copy/paste from [here](https://cloud.google.com/bigquery/docs/reference/v2/jobs/query), but I think we can phrase it better. (use null instead of present) and make it clear this method should be used to check if results should be available.